### PR TITLE
Fix network switching detection

### LINKED ISSUE #2021: [1.30] Token selector is disabled when change network, unknown tokens are in the URL
Related to #1992
1. Open Swap page on Mainnet
2. Do not connect to a wallet
3. Select SYN/AUDIO token pair
4. Change network to Gorli

**AR**: Token selector is disabled 

**ER**: Token selector is enabled, URL is changed to the default state (Sell token = WETH, "Select token' in the To field)

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2021

The bug was introduced here: https://github.com/cowprotocol/cowswap/pull/1942/files#diff-476acfe17c5b8c2a3ba90bcccd7dc4c43ca0b62c5971b172351d2223685ec8b3R64

`chainIdFromProviderWasChanged` was not changed because of it bug, consequently, `chainId` didn't change in URL and the Swap form didn't change. 

I'm very sorry for `useSetupTradeState` hook :( We should refactor it some a day.

Important! This change shouldn't break E2E tests.

### CODE REVIEW

File: src/cow-react/modules/trade/hooks/setupTradeState/useSetupTradeState.ts [hunk 1]
```diff
@@ -62,8 +62,10 @@ export function useSetupTradeState(): void {
   const { state, updateState } = useTradeState()
 
   const chainIdFromUrl = tradeStateFromUrl.chainId
-  // When there is no account from provider, then we consider provider as not connected and use chainId from URL as current
-  const currentChainId = (account ? providerChainId : chainIdFromUrl) || providerChainId
+  // When there is no account from provider (wallet is not connected)
+  // And there is no chainId in provider (edge case in E2E tests)
+  // Then we use chainId from URL as current
+  const currentChainId = (account ? providerChainId : providerChainId || chainIdFromUrl) || providerChainId
```
Comment (tukantje) [inline src/cow-react/modules/trade/hooks/setupTradeState/useSetupTradeState.ts:68]: A bit of a nitpick however why not do this in multiple steps for readability? Something along the lines of;

```
let currentChainId = providerChainId;

if (!account && !currentChainId) {
  currentChainId = chainIdFromUrl;
}
```

Since you want only in one case to use `chainIdFromUrl` and everything else seems to be falling back to `providerChainId`.

**Review by tukantje — APPROVED** [review]

File: src/cow-react/modules/trade/hooks/setupTradeState/useSetupTradeState.ts (hunk 1 above)
Comment (shoom3301) [reply src/cow-react/modules/trade/hooks/setupTradeState/useSetupTradeState.ts]: Thank you!
I don't like using `let` :) Refactored by another way

Comment (tukantje) [reply src/cow-react/modules/trade/hooks/setupTradeState/useSetupTradeState.ts]: Hahah fair enough! Yours is better anyways 🫡

**Review by elena-zh — APPROVED** [review]
Changes LGTM!
I have also connected #1717 issue as the current PR fixes it.
